modules/openfast-library/src/FAST_Subs.f90

```diff
@@ -5153,61 +5581,2048 @@ SUBROUTINE FAST_InitIOarrays( t_initial, p_FAST, y_FAST, m_FAST, ED, BD, SrvD, A
+   TYPE(AeroDyn_Data),       INTENT(INOUT) :: AD                  !< AeroDyn data
+   TYPE(ExtLoads_Data),      INTENT(INOUT) :: ExtLd               !< External loads data
+   TYPE(InflowWind_Data),    INTENT(INOUT) :: IfW                 !< InflowWind data
+   TYPE(ExternalInflow_Data),INTENT(INOUT) :: ExtInfw             !< ExternalInflow data
+   TYPE(SCDataEx_Data),      INTENT(INOUT) :: SC_DX               !< Supercontroller Exchange data
+   TYPE(SeaState_Data),      INTENT(INOUT) :: SeaSt               !< SeaState data
+   TYPE(HydroDyn_Data),      INTENT(INOUT) :: HD                  !< HydroDyn data
+   TYPE(SubDyn_Data),        INTENT(INOUT) :: SD                  !< SubDyn data
+   TYPE(ExtPtfm_Data),       INTENT(INOUT) :: ExtPtfm             !< ExtPtfm_MCKF data
+   TYPE(MAP_Data),           INTENT(INOUT) :: MAPp                !< MAP data
+   TYPE(FEAMooring_Data),    INTENT(INOUT) :: FEAM                !< FEAMooring data
+   TYPE(MoorDyn_Data),       INTENT(INOUT) :: MD                  !< Data for the MoorDyn module
+   TYPE(OrcaFlex_Data),      INTENT(INOUT) :: Orca                !< OrcaFlex interface data
+   TYPE(IceFloe_Data),       INTENT(INOUT) :: IceF                !< IceFloe data
+   TYPE(IceDyn_Data),        INTENT(INOUT) :: IceD                !< All the IceDyn data used in time-step loop
+
+   TYPE(FAST_ModuleMapType), INTENT(INOUT) :: MeshMapData         !< Data for mapping between modules
+
+   INTEGER(IntKi),           INTENT(  OUT) :: ErrStat             !< Error status of the operation
+   CHARACTER(*),             INTENT(  OUT) :: ErrMsg              !< Error message if ErrStat /= ErrID_None
+
+   ! local variables
+   REAL(DbKi)                              :: t_global_next       ! next simulation time (m_FAST%t_global + p_FAST%dt)
+   INTEGER(IntKi)                          :: n_t_global_next     ! n_t_global + 1
+   INTEGER(IntKi)                          :: j_pc                ! predictor-corrector loop counter
+   INTEGER(IntKi)                          :: NumCorrections      ! number of corrections for this time step
+   INTEGER(IntKi), parameter               :: MaxCorrections = 20 ! maximum number of corrections allowed
+   LOGICAL                                 :: WriteThisStep       ! Whether WriteOutput values will be printed
+
+   INTEGER(IntKi)                          :: I, k                ! generic loop counters
+
+   !REAL(ReKi)                              :: ControlInputGuess   ! value of controller inputs
+
+
+   INTEGER(IntKi)                          :: ErrStat2
+   CHARACTER(ErrMsgLen)                    :: ErrMsg2
+   CHARACTER(*), PARAMETER                 :: RoutineName = 'FAST_Solution'
+
+
+   ErrStat = ErrID_None
+   ErrMsg  = ""
+
+   n_t_global_next = n_t_global+1
+   t_global_next = t_initial + n_t_global_next*p_FAST%DT  ! = m_FAST%t_global + p_FAST%dt
+
+   y_FAST%WriteThisStep = NeedWriteOutput(n_t_global_next, t_global_next, p_FAST)
```

Comment:
I'm wondering why this line wasn't included in `FAST_Prework`. 

`y_FAST%WriteThisStep` is used in `FAST_UpdateStates`. While it doesn't matter in this glue code, I assume the reason this routine was split into `FAST_Prework`, `FAST_UpdateStates`, etc, is that the individual routines would be called outside `FAST_Solution`, in which case, `y_FAST%WriteThisStep` would probably not be set properly. 

Is there another reason for keeping it here?

Reply:
Good question.  Since the `AMR-Wind` has the ability to reset OpenFAST back multiple timesteps, it may be that they didn't want to have the write outputs updated during intermediate steps that may be redone. @gantech, do you happen to remember?

Reply:
If they aren't directly setting `y_FAST%WriteThisStep`, then it is probably just using the initial value of `true`. If you don't need file output at every integration step, you will get performance improvements by setting this value.

Reply:
Corrected with #2076 (on the `dev-unstable-pointers` branch which will be merged to `dev` soonish).